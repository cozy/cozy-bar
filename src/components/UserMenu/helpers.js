import { generateWebLink } from 'cozy-client'
import { isFlagshipApp } from 'cozy-device-helper'

const getSettingsLink = ({ client, hash }) => {
  return generateWebLink({
    cozyUrl: client.getStackClient().uri,
    slug: 'settings',
    hash,
    subDomainType: client.getInstanceOptions().subdomain
  })
}

export const openProfileLink = ({ client }) => {
  const link = getSettingsLink({ client, hash: 'profile' })
  window.open(link, '_self')
}

export const openStorageLink = ({ client }) => {
  const link = getSettingsLink({ client, hash: 'storage' })
  window.open(link, '_self')
}

export const logOut = async ({ client, webviewIntent, onLogOut }) => {
  if (onLogOut && typeof onLogOut === 'function') {
    const res = onLogOut()
    if (res instanceof Promise) {
      await res
    }
  } else {
    await client.logout()

    return isFlagshipApp() && webviewIntent
      ? webviewIntent.call('logout')
      : window.location.reload()
  }
}
