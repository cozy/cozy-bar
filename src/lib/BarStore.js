
/* global fetch */

import { Component } from 'react'
import stack from '../lib/stack'
import { create as createIntent } from '../lib/intents'

import CLAUDY_ACTIONS from '../config/claudyActions'
import { EventEmitter } from 'lib/EventEmitter'

const EXCLUDES = ['settings', 'onboarding']
const CATEGORIES = ['cozy', 'partners', 'ptnb']

export default class BarStore extends EventEmitter {
  constructor () {
    super()
    this.claudyActions = null
    this.appsList = [] // all apps, coming soons included
    this.settingsData = null
    // cache
    this.installedApps = [] // to cache already fetched apps icons
    this.helpLink = ''
    this.settingsAppURL = ''
  }

  getClaudyIntent (data) {
    return createIntent(null, 'CLAUDY', 'io.cozy.settings', data)
  }

  getSupportIntent (data) {
    return createIntent(null, 'SUPPORT', 'io.cozy.settings', data)
  }

  async fetchApps () {
    let apps
    try {
      apps = await Promise.all((await stack.get.apps())
      .filter(app => !EXCLUDES.includes(app.attributes.slug))
      .map(async app => {
        const oldApp = this.installedApps.find(item => item.slug === app.attributes.slug)
        let icon

        if (oldApp && oldApp.icon.cached) {
          icon = oldApp.icon
        } else {
          icon = {
            src: await stack.get.icon(app.links.icon),
            cached: true
          }
        }

        return {
          editor: app.attributes.editor,
          name: app.attributes.name,
          slug: app.attributes.slug,
          href: app.links.related,
          category: CATEGORIES.includes(app.attributes.category) ? app.attributes.category : 'others',
          icon
        }
      }))
      this.installedApps = apps
    } catch (e) {
      return {error: e}
    }
    return apps
  }

  fetchComingSoonApps () {
    return stack.get.context()
    .then(context => {
      const comingSoonApps = (context.data && context.data.attributes &&
      context.data.attributes['coming_soon'] &&
      Object.values(context.data.attributes['coming_soon'])) || []

      return comingSoonApps.map(app => {
        let icon

        try {
          icon = app.slug && {
            cached: true,
            src: require(`../assets/icons/comingsoon/icon-${app.slug}.svg`)
          }
        } catch (error) {
          console.warn && console.warn(`Cannot retrieve icon for app ${app.name}:`, error.message)
        }

        return Object.assign({}, app, {
          comingSoon: true,
          icon: icon
        })
      })
    })
    .catch(error => {
      console.warn && console.warn(`Cozy-bar cannot fetch comming soon apps: ${error.message}`)
      return []
    })
  }

  async fetchAppsList () {
    const apps = await this.fetchApps()
    if (apps.error) {
      this.appsList = apps
      return this.appsList
    }
    let comingSoonApps = await this.fetchComingSoonApps()
    comingSoonApps = comingSoonApps
      // drop coming soon apps already installed
      .filter(comingSoonApp => !apps.find(app => app.slug === comingSoonApp.slug))
    this.appsList = apps.concat(comingSoonApps)
    return this.appsList
  }

  shouldEnableClaudy () {
    if (this.claudyActions) return Promise.resolve(this.claudyActions)
    return stack.get.context()
    .then(context => {
      const contextActions = (context.data && context.data.attributes && context.data.attributes['claudy_actions']) || null
      if (!contextActions) return false
      // get an arrays of action
      const claudyActions = contextActions.map(slug => {
        if (CLAUDY_ACTIONS.hasOwnProperty(slug)) {
          // adding also the action slug
          return Object.assign({}, CLAUDY_ACTIONS[slug], { slug })
        }
      }).filter(action => action)
      return claudyActions.length
    })
    .catch(error => {
      console.warn && console.warn(`Cozy-bar cannot fetch Claudy: ${error.message}`)
      return false
    })
  }

  getHelpLink () {
    if (this.helpLink) return Promise.resolve(this.helpLink)
    return stack.get.context()
      .then(context => {
        this.helpLink = (context.data && context.data.attributes && context.data.attributes['help_link']) || null
        return this.helpLink
      })
      .catch(e => {
        console.warn && console.warn('Cannot get Cozy help link')
        return null
      })
  }

  getStorageData () {
    return stack.get.storageData()
      .catch(e => {
        console.warn && console.warn('Cannot get Cozy storage informations')
        return null
      })
  }

  getSettingsAppURL () {
    // If the `settings` app is available, it will used to add the links 'Profile' and 'Connected Devices'
    if (this.settingsAppURL) return Promise.resolve(this.settingsAppURL)
    return stack.get.settingsAppURL()
      .then(settingsAppURL => {
        this.settingsAppURL = settingsAppURL
        return this.settingsAppURL
      })
      .catch(e => {
        console.warn && console.warn('Settings app is unavailable, settings links are disabled')
        return null
      })
  }

  async fetchSettingsData () {
    const storageData = await this.getStorageData()
    const settingsAppURL = await this.getSettingsAppURL()
    const helpLink = await this.getHelpLink()
    this.settingsData = { storageData, settingsAppURL, helpLink }
  }

  logout () {
    stack.logout()
  }
}

export class Provider extends Component {
  getChildContext () {
    return { store: this.store }
  }

  constructor (props, context) {
    super(props, context)
    this.store = props.store
  }

  render ({children}) {
    return (children && children[0]) || null
  }
}
