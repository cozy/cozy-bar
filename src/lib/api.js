import React, { Component } from 'react'

const upperFirstLetter = val => {
  return val[0].toUpperCase() + val.slice(1)
}

/**
 * Wraps argument into a React element if it is a string. Is used
 * for setBar{Left,Right,Center} to be able to pass HTML
 *
 * @param  {[type]} v [description]
 * @return {[type]}   [description]
 */
const wrapInElement = v => {
  if (typeof v === 'string') {
    return <span dangerouslySetInnerHTML={{__html: v}} />
  } else {
    return v
  }
}

/**
 * Creates a React component that enables to access store
 * properties in a declarative way.
 *
 * @param  {BarStore} store
 * @param  {string} attr
 * @return {Component}
 */
const storeComponent = (store, attr) => class extends Component {
  componentDidMount () {
    this.prev = store.get(attr)
    store.set(attr, this.props.children)
  }

  componentWillUnmount () {
    store.set(attr, this.prev)
  }

  componentDidUpdate () {
    store.set(attr, this.props.children)
  }

  render () {
    return null
  }
}

/**
 * Creates a public API from a store
 *
 * - getters/setters for public attributes
 * - React components that act as getters/setters
 *
 * @param  {BarStore} store - Store on which the API will play
 * @return {object} - Methods of the public API
 */
export default store => {
  const contentAttributes = ['barLeft', 'barCenter', 'barRight']
  const methods = {}
  contentAttributes.forEach(attrName => {
    methods['set' + upperFirstLetter(attrName)] = value => store.set(attrName, wrapInElement(value))
    methods[upperFirstLetter(attrName)] = storeComponent(store, attrName)
  })
  return methods
}
