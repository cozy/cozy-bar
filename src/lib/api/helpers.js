export const locations = ['left', 'center', 'right', 'search']

const upperFirstLetter = val => {
  return val[0].toUpperCase() + val.slice(1)
}

export const getJsApiName = location => {
  return `setBar${upperFirstLetter(location)}`
}

export const getReactApiName = location => {
  return `Bar${upperFirstLetter(location)}`
}
