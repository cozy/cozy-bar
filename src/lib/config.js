/* global __TARGET__ */
import deepClone from 'deep-clone'
import deepEqual from 'deep-equal'

import stack from '../lib/stack'

import MENU_CONFIG from '../config/menu'

const EXCLUDES = ['settings', 'onboarding']

async function updateAppsItems (config) {
  let apps

  try {
    apps = await Promise.all((await stack.get.apps())
      .filter(app => !EXCLUDES.includes(app.attributes.slug))
      .map(async app => {
        const oldApp = config.apps.find(item => item.slug === app.attributes.slug)
        let icon

        if (oldApp && oldApp.icon.cached) {
          icon = oldApp.icon
        } else {
          icon = {
            src: await stack.get.icon(app.links.icon),
            cached: true
          }
        }

        return {
          slug: app.attributes.slug,
          l10n: false,
          href: app.links.related,
          icon
        }
      })
    )
  } catch (e) {
    apps = [{error: e}]
  }

  config.apps.length = 0
  Array.prototype.push.apply(config.apps, apps)
}

async function updateDiskUsage (config) {
  let currentDiskUsage

  try {
    currentDiskUsage = await stack.get.diskUsage()
  } catch (e) {
    currentDiskUsage = { error: e.name }
  }

  config.components.storage.currentDiskUsage = currentDiskUsage
}

async function updateDiskQuota (config) {
  let currentDiskQuota

  try {
    currentDiskQuota = await stack.get.diskQuota()
  } catch (e) {
    currentDiskQuota = { error: e.name }
  }

  config.components.storage.currentDiskQuota = currentDiskQuota
}

/**
 * Add / Remove settings' links items regarding the status of
 * the `settings` app
 * @return {Promise}
 */
async function toggleSettingsItems (config) {
  // We reset the settings' links array
  config.subsections.settings.length = 0

  // If the `settings` app is available, we restore links from the root
  // MENU_CONFIG tree, updating the links' URLs with the app URI at same time.
  try {
    await stack.has.settings()
  } catch (e) {
    console.warn('Settings app is unavailable, links are disabled')
    return
  }

  const items = await updateSettingsURIs(MENU_CONFIG.subsections.settings)
  Array.prototype.push.apply(config.subsections.settings, items)
}

/**
 * Replace in the given tree the base URIs for settings' app items
 * @param  {Object}  tree The JSON defined menu entries
 * @return {Promise}      The parsed tree
 */
async function updateSettingsURIs (items) {
  const baseURI = await stack.get.settingsBaseURI()
  return items.map(item => Object.assign({}, item, {href: `${baseURI}#${item.href}`}))
}

/**
 * Clone and parse a root node from a JSON definition tree (aka 'menu')
 * and recursively replace string definitions `_.(group).(entry)` (e.g.
 * `_.components.storage`) with a pointer to the given object in the tree
 * (here, `tree[components][entry]`)
 *
 * @param  {Object} tree                  The tree containing root node and
 *                                        definitions
 * @param  {String} [rootItem='settings'] The root node to parse
 * @return {Object}                       The parsed tree containing pointers
 */
function createMenuPointers (tree) {
  function parse (value, index, array) {
    let path

    if (!value) { return }

    if (Array.isArray(value)) {
      value.forEach(parse)
    } else if (value === Object(value)) {
      Object.keys(value).forEach(key => parse(value[key], key, value))
    } else if (value.match && (path = value.match(/_\.(\w+)(?:\.(\w+))?/i))) {
      if (path[2]) {
        array[index] = clone[path[1]][path[2]]
      } else {
        array[index] = clone[path[1]]
      }
    }
  }

  const clone = deepClone(tree)
  parse(clone)

  return clone
}

/**
 * Helper function to update apps in CONFIG tree
 * @param  {Object}           config the JSON CONFIG tree source
 * @return {Promise(boolean)} a valve that allow to trigger update or not
 */
async function updateApps (config) {
  const oldApps = config.apps.slice()

  await updateAppsItems(config)

  return !deepEqual(oldApps, config.apps)
}

/**
 * Helper function to update all settings related in CONFIG tree
 * @param  {Object}           config the JSON CONFIG tree source
 * @param  {Object}           options
 *                            - storage {Boolean} update the storage component
 *                            - items {Boolean} update settings items list
 * @return {Promise(boolean)} a valve that allow to trigger update or not
 */
async function updateSettings (config, {storage = true, items = true} = {}) {
  let valve = false

  if (__TARGET__ !== 'mobile' && storage) {
    const oldDiskUsage = config.components.storage.currentDiskUsage
    await updateDiskUsage(config)
    await updateDiskQuota(config)
    valve = valve || oldDiskUsage !== config.components.storage.currentDiskUsage
  }

  if (items) {
    const oldSettingsItems = config.subsections.settings.slice()
    await toggleSettingsItems(config)
    valve = valve || !deepEqual(oldSettingsItems, config.subsections.settings)
  }

  return valve
}

export { createMenuPointers, updateSettings, updateApps }
