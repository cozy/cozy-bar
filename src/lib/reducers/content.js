const SET_CONTENT = 'SET_CONTENT'
const UNSET_CONTENT = 'UNSET_CONTENT'

const getLastItemInMap = map => Array.from(map)[map.size - 1]

// action creator
export const setContent = (location, content, id) => ({
  type: SET_CONTENT,
  location,
  content,
  id
})
export const unsetContent = (location, id) => ({
  type: UNSET_CONTENT,
  location,
  id
})

// reducer
export const getDefaultState = () => ({
  left: new Map(),
  center: new Map(),
  right: new Map(),
  search: new Map()
})

export const reducer = (state = getDefaultState(), action) => {
  if (!action.location || typeof action.id === undefined) return state
  switch (action.type) {
    case SET_CONTENT: {
      const currentState = state[action.location]
      currentState.set(action.id, action.content)
      return { ...state, [action.location]: currentState }
    }
    case UNSET_CONTENT: {
      const currentState = state[action.location]
      if (!currentState.get(action.id)) {
        return state
      }
      currentState.delete(action.id)
      return { ...state, [action.location]: currentState }
    }
    default:
      return state
  }
}

// selectors
export const getContent = (state, location) =>
  getLastItemInMap(state[location]) && getLastItemInMap(state[location])[1]
