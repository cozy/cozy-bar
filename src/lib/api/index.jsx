import React, { Component } from 'react'
import { setContent, unsetContent, setLocale, setTheme } from 'lib/reducers'

import { locations, getJsApiName, getReactApiName } from 'lib/api/helpers'

// The React API need unique IDs, so we will increment this variable
let idToIncrement = 0

/**
 * Wraps argument into a React element if it is a string. Is used
 * for setBar{Left,Right,Center} to be able to pass HTML
 *
 * @param  {[type]} v [description]
 * @return {[type]}   [description]
 */
const wrapInElement = v => {
  if (typeof v === 'string') {
    return <span dangerouslySetInnerHTML={{ __html: v }} />
  } else {
    return v
  }
}

/**
 * Creates a React component that enables to access store
 * properties in a declarative way.
 *
 * @param  {BarStore} store
 */
const barContentComponent = (store, location) =>
  class BarContent extends Component {
    componentDidMount() {
      this.componentId = idToIncrement++
      this.setContent(this.props.children)
    }

    setContent(content) {
      try {
        content = React.Children.only(content)
        // eslint-disable-next-line no-empty
      } catch (e) {}
      store.dispatch(setContent(location, content, this.componentId))
    }

    unsetContent() {
      store.dispatch(unsetContent(location, this.componentId))
    }

    componentWillUnmount() {
      this.unsetContent()
    }

    componentDidUpdate(prevProps) {
      if (this.props.children !== prevProps.children) {
        this.setContent(this.props.children)
      }
    }

    render() {
      return null
    }
  }

/**
 * Creates a public API
 *
 * - getters/setters for public attributes
 * - React components that act as getters/setters
 *
 * @param  {ReduxStore} store - Store on which the API will act
 * @return {object} - Methods of the public API
 */
export default store => {
  // setBar{Left,Right,Center} and <Bar{Left,Right,Center} />
  const methods = {}
  locations.forEach(location => {
    /// expose JS API
    methods[getJsApiName(location)] = value =>
      store.dispatch(setContent(location, wrapInElement(value), 'js'))

    // expose React API
    methods[getReactApiName(location)] = barContentComponent(store, location)
  })

  methods.setLocale = (...args) => {
    store.dispatch(setLocale(...args))
  }

  methods.setTheme = (...args) => {
    store.dispatch(setTheme(...args))
  }

  return methods
}
