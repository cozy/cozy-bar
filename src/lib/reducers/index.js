import { combineReducers } from 'redux'
import * as content from 'lib/reducers/content'
import * as locale from 'lib/reducers/locale'
import * as theme from 'lib/reducers/theme'
import appsReducer, * as apps from 'lib/reducers/apps'
import settingsReducer, * as settings from 'lib/reducers/settings'
import contextReducer, * as context from 'lib/reducers/context'

const proxy = (attr, method) => {
  return (state, ...args) => {
    return method(state[attr], ...args)
  }
}

const setContent = content.setContent
const unsetContent = content.unsetContent
const setLocale = locale.setLocale
const setTheme = theme.setTheme
const fetchApps = apps.fetchApps
const setInfos = apps.setInfos
const fetchSettingsData = settings.fetchSettingsData
const logOut = settings.logOut
const fetchContext = context.fetchContext
export {
  setContent,
  unsetContent,
  setLocale,
  setTheme,
  fetchApps,
  setInfos,
  fetchSettingsData,
  logOut,
  fetchContext
}

export const getContent = proxy('content', content.getContent)
export const getLocale = proxy('locale', locale.getLocale)
export const getTheme = proxy('theme', theme.getTheme)
export const getApps = proxy('apps', apps.getApps)
export const getHomeApp = proxy('apps', apps.getHomeApp)
export const isFetchingApps = proxy('apps', apps.isFetchingApps)
export const isCurrentApp = proxy('apps', apps.isCurrentApp)
export const hasFetched = proxy('apps', apps.hasFetched)
export const getStorageData = proxy('settings', settings.getStorageData)
export const getSettingsAppURL = proxy('settings', settings.getSettingsAppURL)
export const isSettingsBusy = proxy('settings', settings.isSettingsBusy)
export const isFetchingSettings = proxy('settings', settings.isFetchingSettings)
export const getHelpLink = proxy('context', context.getHelpLink)
export const getClaudyActions = proxy('context', context.getClaudyActions)
export const shouldEnableClaudy = proxy('context', context.shouldEnableClaudy)

// realtime handlers
export const onRealtimeCreate = apps.receiveApp
export const onRealtimeDelete = apps.deleteApp

export const reducers = {
  content: content.reducer,
  locale: locale.reducer,
  theme: theme.reducer,
  apps: appsReducer,
  settings: settingsReducer,
  context: contextReducer
}

export default combineReducers(reducers)
