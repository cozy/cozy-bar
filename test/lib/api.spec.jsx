import React from 'react'
import { shallow } from 'enzyme'

import createStore from 'lib/store'
import { createBarAPI } from 'lib/api'
import appReducers, { getContent as _getContent } from 'lib/reducers'
import { getDefaultState } from 'lib/reducers/content'
import { locations, getJsApiName, getReactApiName } from 'lib/api/helpers'

const store = createStore()
const api = createBarAPI(store)

// add test reducer to reset state
const RESET_API = 'RESET_API'
const testReducer = (state, action) => {
  if (action.type === RESET_API) {
    const newState = Object.assign({}, state, {
      content: getDefaultState()
    })
    return newState
  }
  return appReducers(state, action)
}

store.replaceReducer(testReducer)

const getContent = location => {
  return _getContent(store.getState(), location)
}

const contentToTest = {
  left: 'Left',
  center: <p>Center</p>,
  right: (
    <div>
      test3
      <span>Right</span>
    </div>
  ),
  search: 'Search'
}

describe('api spec', function() {
  beforeEach(() => {
    store.dispatch({
      type: RESET_API
    })
  })

  it('should set content of the store via imperative api', function() {
    locations.map(location => {
      api[getJsApiName(location)](contentToTest[location])
      expect(getContent(location)).toMatchSnapshot()
    })
  })

  it('should set content of the store via declarative api', function() {
    locations.map(location => {
      const BarAPI = api[getReactApiName(location)]
      shallow(
        <BarAPI>
          <div>{contentToTest[location]}</div>
        </BarAPI>
      )
      expect(getContent(location)).toMatchSnapshot()
    })
  })

  it('should replace content of the store instead of adding new entry', function() {
    locations.map(location => {
      const BarAPI = api[getReactApiName(location)]
      const barComponent = shallow(
        <BarAPI>
          <div>{contentToTest[location]}</div>
        </BarAPI>
      )
      const initialContent = getContent(location)
      // update content
      expect(initialContent).toEqual(getContent(location))
      expect(store.getState().content[location].size).toBe(1)
      barComponent.setProps({
        children: <div>Left Updated</div>
      })
      // content has correctly be updated
      expect(initialContent).not.toEqual(getContent(location))
      // but no new content added, it was replaced in the state
      expect(store.getState().content[location].size).toBe(1)
    })
  })

  it('should unset content of the store on unmount', function() {
    locations.map(location => {
      const BarAPI = api[getReactApiName(location)]
      const barComponent = shallow(
        <BarAPI>
          <div>{contentToTest[location]}</div>
        </BarAPI>
      )
      expect(store.getState().content[location].size).toBe(1)
      // unmount
      barComponent.unmount()
      expect(store.getState().content[location].size).toBe(0)
    })
  })

  it('should set locale', function() {
    const { setLocale } = api
    // default lang is `en`
    expect(store.getState().locale).toBe('en')
    setLocale('fr')
    expect(store.getState().locale).toBe('fr')
    setLocale('de')
    expect(store.getState().locale).toBe('de')
  })

  it('should set theme', function() {
    const { setTheme } = api
    // default theme is `default`
    expect(store.getState().theme.name).toBe('default')
    expect(store.getState().theme.overrides).toEqual({})
    setTheme('primary')
    expect(store.getState().theme.name).toBe('primary')
    expect(store.getState().theme.overrides).toEqual({})
    setTheme('primary', { primaryColor: 'red' })
    expect(store.getState().theme.name).toBe('primary')
    expect(store.getState().theme.overrides).toEqual({ primaryColor: 'red' })
    setTheme('wrongTheme')
    expect(store.getState().theme.name).toBe('default')
    expect(store.getState().theme.overrides).toEqual({})
  })

  it('should set webviewContext', function() {
    const { setWebviewContext } = api
    // default webviewContext is `undefined`
    expect(store.getState().unserializable.webviewContext).toBe(undefined)
    setWebviewContext('foo')
    expect(store.getState().unserializable.webviewContext).toBe('foo')
    setWebviewContext('bar')
    expect(store.getState().unserializable.webviewContext).toBe('bar')
    setWebviewContext(undefined)
  })
})
