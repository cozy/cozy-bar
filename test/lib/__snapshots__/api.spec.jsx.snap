// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`api spec should set content of the store via declarative api 1`] = `
<div>
  Left
</div>
`;

exports[`api spec should set content of the store via declarative api 2`] = `
<div>
  <p>
    Center
  </p>
</div>
`;

exports[`api spec should set content of the store via declarative api 3`] = `
<div>
  <div>
    test3
    <span>
      Right
    </span>
  </div>
</div>
`;

exports[`api spec should set content of the store via imperative api 1`] = `
<span
  dangerouslySetInnerHTML={
    Object {
      "__html": "Left",
    }
  }
/>
`;

exports[`api spec should set content of the store via imperative api 2`] = `
<p>
  Center
</p>
`;

exports[`api spec should set content of the store via imperative api 3`] = `
<div>
  test3
  <span>
    Right
  </span>
</div>
`;
